packages/dd-trace/src/appsec/iast/iast-context.js

```diff
@@ -0,0 +1,42 @@
+const { storage } = require('../../../../datadog-core')
+
+const IAST_CONTEXT_KEY = Symbol('_dd.iast.context')
+
+function getIastContext (store) {
+  const _store = store || storage.getStore()
+  return _store && _store[IAST_CONTEXT_KEY]
+}
+
+function saveIastContext (store, topContext, context) {
```

Comment:
Initially, we had the intention to use only "storage" to save the iast context. To get the `topContext` we need the request and it is not always available.

But we discovered that we need to subscribe to the request `close` event, we had seen that there `storage` is not always available (e.g. when the browser cancel the request, close event is launched, but store is not available)

This is why we need to use store and topContext.

Reply:
Not sure I understand. The context is on the store, so if the store is not available wouldn't the context have the same issue? It also sounds like we're hiding a separate issue if the store is not available when it should be in the first place.

Reply:
We are getting the `topContext` from `web.getContext`, it reads the context from a `WeakMap<request, context>` (https://github.com/DataDog/dd-trace-js/blob/291eec7a6887af82336a910facf2dd7dc8758eec/packages/dd-trace/src/plugins/util/web.js#L355)

If the store is not available (sometimes on request close) and request is available, we can get it from the request.

This is the reason to save iast related context in store and in web root context object.

I could only play this behaviour with the multiple F5 issue. As we discussed yesterday, I will try to fix it in this PR, if it works, I will be able to simplify it.

Reply:
This is definitely an issue outside of IAST as well, where the store needs to always be available. If we need a separate mechanism there is definitely some underlying issue that needs fixing, but it might be enough to add a TODO if it's not doable right now and we can look into it later. This is likely to happen soon anyway since the existence of the aforementioned WeakMap is specifically that same problem but for tracing.

Reply:
Ok, If I see that I can fix it by unifying `finish` and `close` event, I will fix it. If I can't, I will write a TODO

- [ ] Fix or write a TODO to use store and not the `WeakMap`

Reply:
If the close event is generated from an unexpected `close` the `store` is new, is not related with the opened request store. I don't know how can we fix it without the mentioned `WeakMap`. So I have write a TODO to change it in the future, but we still using `web.getContext()`

Reply:
Doesn't look like the TODO is there?

Reply:
I have written the TODO where `cleanIastContext` is called, but yes, I'm going to add it also here